### ISSUE TITLE
Show recommended extensions when using live share

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#21409]
Currently extensions needed for a project can be exposed to the user via `.vscode/extensions.json`. But when live sharing a session that folder is not shared, and recommended extensions are not exposed to the participants. This is fairly inconvenient when trying to share in eg. [vscode.dev](https://vscode.dev) where only the default extensions are installed and the participant will need to go search for and install eg. syntax highlighting every single session.

In summary: extend the recommended extensions functionality to live share participants.
